include/translated-term.php

```diff
@@ -338,4 +338,181 @@ protected function get_db_infos() {
+	 *     @type int          $parent       The id of the parent term. Default 0.
+	 *     @type string       $slug         The term slug to use. Default empty string.
+	 *     @type PLL_Language $lang         The term language object.
+	 *     @type string[]     $translations The translation group to assign to the term with language slug as keys and `term_id` as values.
+	 * }
+	 * @return array|WP_Error An array containing the `term_id` and `term_taxonomy_id`,
+	 *                        WP_Error otherwise.                  WP_Error otherwise.
+	 */
+	public function update( int $term_id, array $args = array() ) {
+		$term = get_term( $term_id );
+		if ( ! $term instanceof WP_Term ) {
+			return new WP_Error( 'invalid_term', __( 'Empty Term.', 'polylang' ) );
+		}
+
+		$new_language = ! empty( $args['lang'] ) ? $this->model->get_language( $args['lang'] ) : null;
+		if ( false === $new_language ) {
+			return new WP_Error( 'invalid_language', __( 'Please provide a valid language.', 'polylang' ) );
+		}
+
+		$old_language = $this->get_language( $term_id );
+		if ( false === $old_language ) {
+			return new WP_Error( 'broken_language', __( 'Given term has a broken language.', 'polylang' ) );
+		}
+
+		if ( is_null( $new_language ) ) {
+			$new_language = $old_language;
+		}
+
+		$language_callback = $this->get_slug_management_language_callback( $new_language );
+		$parent_callback   = $this->get_slug_management_parent_callback( $args['parent'] ?? $term->parent );
+
+		$this->toggle_slug_management( $language_callback, $parent_callback );
+		$term = wp_update_term( $term->term_id, $term->taxonomy, $args );
+		$this->toggle_slug_management( $language_callback, $parent_callback );
+
+		if ( $old_language->slug !== $new_language->slug ) {
+			$this->set_language( $term_id, $new_language );
+		}
+
+		if ( ! empty( $args['translations'] ) ) {
+			$this->model->term->save_translations(
+				$term_id,
+				array_merge(
+					$args['translations'],
+					array(
+						$new_language->slug => $term_id,
+					)
+				)
+			);
+		}
+
+		return $term;
+	}
+
+	/**
+	 * Toggles Polylang term slug management.
+	 * Must be used before and after any term slug modification or insertion.
+	 *
+	 * @since 3.7
+	 *
+	 * @param callable $language_callback The language callback for term slug modification.
+	 * @param callable $parent_callback   The parent callback for term slug modification.
+	 * @return void
+	 */
+	private function toggle_slug_management( $language_callback, $parent_callback ): void {
+		static $enabled = false;
+
+		if ( $enabled ) {
+			// Clean up!
+			remove_filter( 'pll_inserted_term_language', $language_callback );
+			remove_filter( 'pll_inserted_term_parent', $parent_callback );
+			$enabled = false;
+		} else {
+			// Set term parent and language for suffixed slugs.
+			add_filter( 'pll_inserted_term_language', $language_callback );
+			add_filter( 'pll_inserted_term_parent', $parent_callback );
+			$enabled = true;
```

Comment:
You can have a common instruction with `$enabled = ! $enabled`.

Reply:
I must say I'm not very fond of this.
With `$enabled = false` or `$enabled = true` I can directly tell what value is assigned.
Also, toggle function might already be quite confusion as it is, I suggest to keep it.
Do you agree?